pkg/util/gendoc/gendoc.go

```diff
@@ -0,0 +1,143 @@
+package gendoc
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/spf13/pflag"
+)
+
+const (
+	gendocTypeFlag = "type"
+
+	gendocMarkdown = "md"
+	gendocMan      = "man"
+
+	depthFlag     = "depth"
+	extensionFlag = "extension"
+)
+
+// Command returns command which generates user documentation for the argument.
+func Command(rootCmd *cobra.Command) *cobra.Command {
+	gendocCmd := &cobra.Command{
+		Use:   "gendoc <dir>",
+		Short: "Generate documentation for this command.",
+		Long: `Generate documentation for this command. If the template is not provided,
+builtin cobra generator is used and each subcommand is placed in
+a separate file in the same directory.
+
+The last optional argument specifies the template to use with text/template.
+In this case there is a number of helper functions which can be used:
+  replace STR FROM TO -- same as strings.ReplaceAll
+  join ARRAY SEPARATOR -- same as strings.Join
+  split STR SEPARATOR -- same as strings.Split
+  fullUse CMD -- slice of all command names starting from the parent
+  listFlags CMD -- list of command flags
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				_ = cmd.Usage()
+				os.Exit(1)
+			}
+
+			err := os.MkdirAll(args[0], os.ModePerm)
+			if err != nil {
+				return fmt.Errorf("can't create directory: %w", err)
+			}
+
+			if len(args) == 2 {
+				data, err := ioutil.ReadFile(args[1])
+				if err != nil {
+					return fmt.Errorf("can't read the template '%s': %w", args[1], err)
+				}
+
+				return generateTemplate(cmd, rootCmd, args[0], data)
+			}
+
+			typ, _ := cmd.Flags().GetString(gendocTypeFlag)
+			switch typ {
+			case gendocMarkdown:
+				return doc.GenMarkdownTree(rootCmd, args[0])
+			case gendocMan:
+				hdr := &doc.GenManHeader{
+					Section: "1",
+					Source:  "NSPCC & Morphbits",
+				}
+				return doc.GenManTree(rootCmd, hdr, args[0])
+			default:
+				return fmt.Errorf("type must be 'md' or 'man'")
+			}
+		},
+	}
+
+	ff := gendocCmd.Flags()
+	ff.StringP(gendocTypeFlag, "t", gendocMarkdown, "type for the documentation ('md' or 'man')")
```

Comment:
Consider using constants.

Reply:
We have 2 and they are small, let's leave it compact.